File: Second Year/Semester II/Web Programming/lab9_node/src/services/hotelService.js
const hotelRepository = require('../repositories/hotelRepository');

function toDto(hotel) {
    return {
        id: hotel.id,
        name: hotel.name,
        address: hotel.address
    };
}

async function getHotels(name, address) {
    const hotels = await hotelRepository.getHotels(name, address);
    return hotels.map(toDto);
}

async function insertHotel(name, address) {
    const idCreatedHotel = await hotelRepository.insertHotel(name, address);
    const createdHotel = await hotelRepository.getHotelById(idCreatedHotel);
    return toDto(createdHotel);
}

async function updateHotel(id, name, address) {
    const updatedHotel = await hotelRepository.updateHotel(id, name, address);
    if (!updatedHotel) {
        return null;
    }
    const hotel = await hotelRepository.getHotelById(id);
    return toDto(hotel);
}

async function removeHotel(id) {
    return await hotelRepository.removeHotel(id);
}

module.exports = {
    getHotels,
    insertHotel,
    updateHotel,
    removeHotel
};

File: Second Year/Semester II/Web Programming/lab8_aspnet/lab8_aspnet/Services/HotelService.cs
using lab8_aspnet.DTOs;
using lab8_aspnet.Models;
using lab8_aspnet.Repositories.Interfaces;
using lab8_aspnet.Services.Interfaces;

namespace lab8_aspnet.Services
{
    public class HotelService : IHotelService
    {
        private readonly IHotelRepository _hotelRepository;

        public HotelService(IHotelRepository hotelRepository)
        {
            _hotelRepository = hotelRepository;
        }

        private static HotelDto ToDto(Hotel hotel) => new HotelDto
        {
            Id = hotel.Id,
            Name = hotel.Name,
            Address = hotel.Address
        };

        public async Task<List<HotelDto>> GetHotelsAsync(string? name, string? address) =>
            (await _hotelRepository.GetHotelsAsync(name, address)).Select(ToDto).ToList();

        public async Task<HotelDto> InsertHotelAsync(HotelCreateRequest request)
        {
            var hotel = new Hotel
            {
                Name = request.Name,
                Address = request.Address
            };

            var createdHotel = await _hotelRepository.InsertHotelAsync(hotel);
            return ToDto(createdHotel);
        }

        public async Task<HotelDto?> UpdateHotelAsync(int id, HotelCreateRequest request)
        {
            var updatedHotel = await _hotelRepository.UpdateHotelAsync(id, request.Name, request.Address);
            return updatedHotel == null ? null : ToDto(updatedHotel);
        }

        public async Task<bool> DeleteHotelAsync(int id) =>
            await _hotelRepository.DeleteHotelAsync(id);
    }
}